src/components/ComputeModal.js

```diff
@@ -679,18 +712,29 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
   const renderActionButton = () => {
     const { runtime: existingRuntime, hasGpu } = getExistingEnvironmentConfig()
     const { runtime: desiredRuntime } = getDesiredEnvironmentConfig()
-    const commonButtonProps = hasGpu && viewMode !== 'deleteEnvironmentOptions' ?
-      { disabled: true, tooltip: 'Cloud compute with GPU(s) cannot be updated. Please delete it and create a new one.' } :
-      { disabled: !hasChanges() || !!errors, tooltip: Utils.summarizeErrors(errors) }
-    const canShowCustomImageWarning = viewMode === undefined
+    const commonButtonProps = Utils.cond([
+      hasGpu && viewMode !== 'deleteEnvironmentOptions',
+      () => ({ disabled: true, tooltip: 'Cloud compute with GPU(s) cannot be updated. Please delete it and create a new one.' })
+    ], [
+      computeConfig.gpuEnabled && _.isEmpty(getValidGpuTypesForZone(computeConfig.computeZone)) && viewMode !== 'deleteEnvironmentOptions',
+      () => ({ disabled: true, tooltip: 'GPUs not available in this location.' })
+    ], [
+      currentPersistentDiskDetails && currentPersistentDiskDetails.zone.toUpperCase() !== computeConfig.computeZone && viewMode !== 'deleteEnvironmentOptions',
```

Comment:
We prefer explicit casting to boolean when a non-boolean const is used in logical operations:

```suggestion
      !!currentPersistentDiskDetails && currentPersistentDiskDetails.zone.toUpperCase() !== computeConfig.computeZone && viewMode !== 'deleteEnvironmentOptions',
```

Reply:
Done